plugins/modules/foreman_image.py

```diff
@@ -0,0 +1,128 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with This program.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: foreman_image
+short_description: Manage Foreman Images using Foreman API
+description:
+  - Create and Delete Foreman Images using Foreman API
+author:
+  - "Mark Hlawatschek (@hlawatschek) ATIX AG"
+options:
+  name:
+    description: Image name
+    required: true
+    type: str
+  compute_resource:
+    description: Compute resource the image is assigned to
+    required: true
+    type: str
+  uuid:
+    aliases:
+      - image_uuid
+    description: UUID or Marketplace URN of the operatingsystem image
+    required: true
+    type: str
+  image_username:
+    description: Username that is used to login into the operating system
+    required: true
+    type: str
+  image_password:
+    description: Password that is used to login into the operating system
+    required: false
+    type: str
+  operatingsystem:
+    description: Operating system that will be deployed using the image
+    required: true
+    type: str
+  architecture:
+    description: architecture of the image
+    required: true
+    type: str
+  user_data:
+    description: Image supports user_data
+    required: false
+    type: bool
+extends_documentation_fragment:
+  - foreman
+  - foreman.entity_state
+'''
+
+EXAMPLES = '''
+   - name: create Image for EC2
+     foreman_image:
+        name: CentOS
+        image_uuid: "ami-0ff760d16d9497662"
+        image_username: "centos"
+        operatingsystem: "CentOS 7"
+        compute_resource: "AWS"
+        architecture: "x86_64"
+'''
+
+RETURN = ''' # '''
+
+from ansible.module_utils.foreman_helper import ForemanEntityAnsibleModule
+
+
+class ForemanImageModule(ForemanEntityAnsibleModule):
+    pass
+
+
+def main():
+    module = ForemanImageModule(
+        argument_spec=dict(
+            image_username=dict(required=True),
+            image_password=dict(no_log=True),
+        ),
+        entity_spec=dict(
+            name=dict(required=True),
+            username=dict(type='invisible'),
+            uuid=dict(required=True, aliases=['image_uuid']),
+            password=dict(type='invisible', no_log=True),
+            compute_resource=dict(type='entity', required=True),
+            architecture=dict(type='entity', required=True),
+            operatingsystem=dict(type='entity', required=True),
+            user_data=dict(type='bool')
+        ),
+        entity_scope="compute_resource"
```

Comment:
```suggestion
        entity_scope="compute_resource",
        entity_resolve=False,
```

Reply:
after adding the changes, somehow entity_dict['compute_resource']['id'] was not set. Therefore I added a manual search for the compute resource.